src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrNodeResource.java

```diff
@@ -227,6 +212,31 @@ private InputStream getInputStream() {
         return null;
     }
 
+    static @NotNull Property getPrimaryProperty(@NotNull Node node) throws RepositoryException {
```

Comment:
This code is cleaner than before patch, but for the purposes of introducing a URIProvider implementation I dont think its is necessary, unless I missed something.

I would recommend that these changes are done in a separate PR if they are considered to have benefit worthy of making the change.

Reply:
This has been refactored because this method is used from the new URIProvider.

Reply:
I think that is wrong, but its not for me to say.